src/App.tsx

```diff
@@ -1,56 +1,72 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './DropdownMenu/DropdownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const timerId = useRef(0);
+
+  const handleQueryChange = (event:React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+
+    window.clearTimeout(timerId.current);
+    timerId.current = window.setTimeout(() => {
+      setAppliedQuery(event.target.value);
+    }, 500);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => {
+      const readyQuery = appliedQuery.toLowerCase().trim();
+      const personName = person.name.toLocaleLowerCase();
+
+      return personName.includes(readyQuery);
+    });
+  }, [appliedQuery]);
+
+  const onSubmit = useMemo(() => {
+    return (person:Person) => {
+      const { name } = person;
+
+      setQuery(name);
+      setSelectedPerson(person);
+      setIsFocused(false);
+    };
+  }, []);
+
+  const isSelectedPerson = query === selectedPerson?.name;
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {isSelectedPerson
+          ? (`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`)
+          : ('No selected person')}
```

Comment:
Demo not updated.
<img width="418" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/54860641/f9c08c9e-ca6d-486a-93ba-b2ef004d030a">
<img width="1352" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/54860641/dc279734-e4fc-4d91-9ada-45e6699688f7">

Reply:
I am sorry. I fixed it.